ironic_prometheus_exporter/parsers/ipmi.py

```diff
@@ -0,0 +1,169 @@
+            label = '_'.join(e[:-1]).replace('-', '_')
+
+        if extract_unit:
+            sensor_read = payload[entry]['Sensor Reading'].split()
+            if len(sensor_read) > 1:
+                sufix = '_' + sensor_read[-1].lower()
+
+        if special_label == 'memory':
+            if 'mem' not in label and 'memory' not in label:
+                label = 'memory_' + label
+
+        metric_name = prefix + label + sufix
+        if metric_name in metric_dic:
+            metric_dic[metric_name].append(entry)
+        else:
+            metric_dic[metric_name] = [entry]
+    return metric_dic
+
+
+def extract_labels(entries, payload, node_name):
+    """ This function extract the labels to be used by a metric
+
+    If a metric has many entries we add the 'Sensor ID' information as label
+    otherwise we will only use the default label that is the node name.
+
+    e.g: for Temperature we have two entries for baremetal_temperature_celcius
+    metric ('Temp (0x1)' and 'Temp (0x2)') and one entry for 'Inlet Temp (0x5)'
+    and other for 'Exhaust Temp (0x6)', this will produce a dictionary where
+    the keys are the entries and the values are the respective label to be used
+    when writting the metrics in the Prometheus format.
+    {'Inlet Temp (0x5)': '{node_name=...}',
+     'Exhaust Temp (0x6)': '{node_name=...}',
+     'Temp (0x1)': '{node_name=...,sensor=Temp1}',
+     'Temp (0x2)': '{node_name=...,sensor=Temp2}'}
+
+    """
+    default_label = 'node_name="%s"' % node_name
+    if len(entries) == 1:
+        return {entries[0]: '{%s}' % default_label}
+    entries_labels = {}
+    for entry in entries:
+        try:
+            sensor = payload[entry]['Sensor ID'].split()
+            sensor_id = str(int(re.sub(r'[\(\)]', '', sensor[-1]), 0))
+            metric_label = [default_label,
+                            'sensor="%s"' % (sensor[0] + sensor_id)]
+            entries_labels[entry] = '{%s}' % ','.join(metric_label)
+        except Exception as e:
+            LOG.error(e)
+    return entries_labels
+
+
+def extract_values(entries, payload, use_ipmi_format=True):
+    values = {}
+    for entry in entries:
+        try:
+            no_values = ['No Reading', 'Disabled']
+            if payload[entry]['Sensor Reading'] in no_values:
+                values[entry] = None
+            else:
+                sensor_values = payload[entry]['Sensor Reading'].split()
+                if not use_ipmi_format:
+                    if not re.search(r'(\d+(\.\d*)?|\.\d+)', sensor_values[0]):
+                        raise Exception("No valid value in Sensor Reading")
+                    values[entry] = sensor_values[0]
+                if len(sensor_values) > 1:
+                    values[entry] = sensor_values[0]
+                elif sensor_values[0] == "0h":
+                    values[entry] = 0
+                else:
+                    values[entry] = 1
+        except Exception as e:
+            LOG.error(e)
```

Comment:
This is a recipe for a debugging nightmare. At the very least you need to use `LOG.exception` so that the logs actually contain some context. Keep in mind that e.g. the message of a `KeyError` is just a key.

Reply:
The same applies to other `extract_labels`.

Reply:
I will change and also add more logs information to help with the debug